libs/librepcb/common/widgets/tabwidget.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * LibrePCB - Professional EDA for everyone!
+ * Copyright (C) 2013 LibrePCB Developers, see AUTHORS.md for contributors.
+ * https://librepcb.org/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef LIBREPCB_TABWIDGET_H
+#define LIBREPCB_TABWIDGET_H
+
+/*******************************************************************************
+ *  Includes
+ ******************************************************************************/
+#include <QtCore>
+#include <QtWidgets>
+
+/*******************************************************************************
+ *  Namespace / Forward Declarations
+ ******************************************************************************/
+namespace librepcb {
+
+/*******************************************************************************
+ *  Class TabWidget
+ ******************************************************************************/
+
+/**
+ * @brief A QTabWidget subclass that allows closing tabs with the middle mouse
+ * button.
+ *
+ * This subclass of QTabWidget overrides the event filter. If it detects a
+ * middle mouse click on a tab that is closable and where the tab button is not
+ * null, it emits the tabCloseRequested event.
+ *
+ * @author dbrgn
+ * @date 2018-12-07
+ */
+class TabWidget final : public QTabWidget {
+  Q_OBJECT
+
+public:
+  TabWidget(QWidget* parent);
```

Comment:
Add `explicit` (to avoid surprises), `noexcept` (according our coding guidelines) and `= nullptr` (for consistency and to support default-constructing):
```suggestion
  explicit TabWidget(QWidget* parent = nullptr) noexcept;
```

In addition, please explicitly disable the copy constructor and assignment operator (see for example the `StatusBar` class).